src/internal/clusterstate/v2.8.0/collection.go

```diff
@@ -239,3 +239,66 @@ func migratePostgreSQLCollection(ctx context.Context, tx *pachsql.Tx, name strin
 	}
 	return nil
 }
+
+func setupPostgresCollections(ctx context.Context, sqlTx *pachsql.Tx, collections ...*postgresCollection) error {
+	for _, col := range collections {
+		columns := []string{
+			"createdat timestamp with time zone default current_timestamp",
+			"updatedat timestamp with time zone default current_timestamp",
+			"proto bytea",
+			"version text",
+			"key text primary key",
+		}
+
+		indexFields := []string{"'key'"}
+		for _, idx := range col.indexes {
+			name := indexFieldName(idx)
+			columns = append(columns, name+" text")
+			indexFields = append(indexFields, "'"+name+"'")
+		}
+
+		log.Info(ctx, fmt.Sprintf("Creating collections.%s table", col.table))
+		createTable := fmt.Sprintf("create table collections.%s (%s);", col.table, strings.Join(columns, ", "))
+		if _, err := sqlTx.Exec(createTable); err != nil {
+			return errors.EnsureStack(err)
+		}
+
+		for _, idx := range col.indexes {
+			createIndex := fmt.Sprintf("create index on collections.%s (%s);", col.table, indexFieldName(idx))
+			if _, err := sqlTx.Exec(createIndex); err != nil {
+				return errors.EnsureStack(err)
+			}
+		}
+
+		updatedatTrigger := fmt.Sprintf(`
+	create trigger updatedat_trigger
+		before insert or update or delete on collections.%s
+		for each row execute procedure collections.updatedat_trigger_fn();
+	`, col.table)
+		if _, err := sqlTx.ExecContext(ctx, updatedatTrigger); err != nil {
+			return errors.EnsureStack(err)
+		}
+
+		notifyTrigger := fmt.Sprintf(`
+	create trigger notify_trigger
+		after insert or update or delete on collections.%s
+		for each row execute procedure collections.notify_trigger_fn(%s);
+	`, col.table, strings.Join(indexFields, ", "))
+		if _, err := sqlTx.ExecContext(ctx, notifyTrigger); err != nil {
+			return errors.EnsureStack(err)
+		}
+	}
+	return nil
+}
+
+// newPostgresCollection creates a new collection backed by postgres.
+func newPostgresCollection(name string, indexes []*index, opts ...colOption) *postgresCollection {
```

Comment:
if the only caller here is just taking name and not indexes and opts arguments, maybe we should remove all the associated logic

Reply:
Done.